src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/RazorSyntaxTreeExtensions.cs

```diff
@@ -34,5 +34,20 @@ public static IReadOnlyList<TagHelperSpanInternal> GetTagHelperSpans(this RazorS
 
             return visitor.TagHelperSpans;
         }
+
+        /// <summary>
+        /// Gets top-level and nested diagnostics from the <paramref name="syntaxTree"/>.
+        /// </summary>
+        /// <param name="syntaxTree">The syntax t ree.</param>
+        /// <returns>The list of <see cref="RazorDiagnostic"/>s.</returns>
+        public static IReadOnlyList<RazorDiagnostic> GetAllDiagnostics(this RazorSyntaxTree syntaxTree)
+        {
+            if (syntaxTree == null)
+            {
+                throw new ArgumentNullException(nameof(syntaxTree));
+            }
+
+            return syntaxTree.Root.GetAllDiagnostics();
```

Comment:
I don't super get why we need this **and** `.Diagnostics`. This is a subset of `.Diagnostics`.

Reply:
Ya totally agree with you. Added this before I updated RazorSyntaxTree's diagnostics to encompass all diagnostics. An oversight